netdata/kernel-collector
# Convert sync eBPF programs to CO-RE

### Description

This PR is extending CO-RE bringing monitoring for 4 additional `sync` syscalls.
It also adds to `sync` tester the possibility to load either `kprobe` or `trampoline`.

### Test


1. Clone branch
2. Run `./tools/check-kernel-core.sh` to test your kernel. If you do not receive any warning, your host has support for BTF.
3. Install `bpftool` on your host.
4. Execute commands: `cd co-re && make`
5. Finally run: `./tests/sync`. If you do not have any warning, and you are able to see the following output:

```bash
bash-5.1# ./tests/sync --probe
__x64_sys_syncfs: probe loaded with success
__x64_sys_msync: probe loaded with success
__x64_sys_sync_file_range: probe loaded with success
__x64_sys_fsync: probe loaded with success
__x64_sys_fdatasync: probe loaded with success
bash-5.1# ./tests/sync --trampoline
__x64_sys_syncfs: entry loaded with success
__x64_sys_msync: entry loaded with success
__x64_sys_sync_file_range: entry loaded with success
__x64_sys_fsync: entry loaded with success
__x64_sys_fdatasync: entry loaded with success
bash-5.1# 
```

everything is running as expected.
